### ISSUE TITLE
Background color is not computed correctly when there are multiple layers of non-opaque background colors

### ISSUE THREAD

**Issue by heryandi** [dequelabs/axe-core#2924]
Product: axe-core

Expectation: I expect axe-core to use the right background color value (as can be observed by a color-picker) for color contrast calculation.

Actual: axe-core doesn't seem to calculate the right background color. Seems like it's only wrong when there are multiple layers of non-opaque background colors involved.

Motivation: This should be fixed so the correct color contrast value can be computed.

<pre>
    <code>axe-core version: 4.2.0</code>
</pre>

Sample test cases:
- Case 1
```html
<html>
    <body>
        <div style="background-color: red;">
            <!-- 'blue' with 0.3 alpha -->
            <div style="background-color: rgba(0, 0, 255, 0.3);">
                <!-- 'green' with 0.3 alpha -->
                <div style="background-color: rgba(0, 128, 0, 0.3);">
                    <p>Some text</p>
                </div>
            </div>
        </div>
    </body>
</html>
```
Expected background color (as picked up by a color-picker): #7C2736
Background-color computed by axe-core (`bgColor` inside the `data` field): #7D145B

- Case 2

```html
<html>
    <body>
        <div style="background-color: rgba(255, 255, 255, 1.0);">
            <!-- 'green' with 0.25 alpha -->
            <div style="background-color: rgba(0, 128, 0, 0.25);">
                <!-- 'red' with 0.5 alpha -->
                <div style="background-color: rgba(255, 0, 0, 0.5);">
                    <p>Some text</p>
                </div>
            </div>
        </div>
    </body>
</html>
```

Expected background color (as picked up by a color-picker): #DF6F5F
Background-color computed by axe-core (`bgColor` inside the `data` field): #AF8860

I think (but not sure) this is because the [flattenColors function](https://github.com/dequelabs/axe-core/blob/fedfd72a6e9174c2cb551103ad82839610a8db7d/lib/commons/color/flatten-colors.js#L12-L20) doesn't apply the [simple alpha compositing formula](https://www.w3.org/TR/compositing-1/#simplealphacompositing).

I am using a Chrome extension for my color-picker, if that matters: https://chrome.google.com/webstore/detail/colorpick-eyedropper/ohcpnigalekghcmgcdcenkpelffpdolg

**Comment by WilcoFiers** [comment]
Thank you for reporting @heryandi. I can't say I've ever checked the accuracy of the flattenColor function, but I can see you're right about it not being accurate. I'll definitely look into this.

**Comment by dylanb** [comment]
Can we see code pen for the HTML and CSS for this? I get all sorts of different colors uing the OS X built-in color sampling app. None of them match the reported colors.

**Comment by heryandi** [comment]
Hi, here are the codepen links for those two example cases:
* Case 1: https://codepen.io/heryandi/pen/qBmBxmG
* Case 2: https://codepen.io/heryandi/pen/ExmxQvG

No CSS needed since those examples use only inline CSS.

I tried using [gpick](http://www.gpick.org/) instead of the previous Chrome extension, but the colors picked up are exactly the same as the Chrome extension.

**Comment by jribbens** [comment]
Here's another real-world case of what seems to be the same issue: https://www.bracknell-forest.gov.uk/health-and-social-care/dementia-support
axe is complaining that some of the text is definitely bad contrast when in reality it is #333 on #fff .

**Comment by straker** [comment]
I believe I have determined the issue. When blending multiple colors, our code finds the visual stack order of the elements (in this case it would be: `[p, div#red, div#green, div#white]`) and then [flattens the colors in that order (top down)](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/color/get-background-color.js#L79-L81). However, this produces the incorrect result as doing so with the provided html produces `rgba(175.3125, 105.625, 95.625, 1)`:

![image](https://user-images.githubusercontent.com/2433219/136114758-8da71099-73d1-45ae-84b2-be27586b26ba.png)

Instead, we need to flatten the colors from the bottom up, which results in the correct final color `rgba(223.125, 111.625, 95.625, 1)`

![image](https://user-images.githubusercontent.com/2433219/136114875-ac220387-a3fb-4aaa-8d91-02144acd44cc.png)
